agent/engine/docker_task_engine.go

```diff
@@ -1133,15 +1133,22 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 	}
 
 	if execcmd.IsExecEnabledContainer(container) {
-		tId, err := task.GetID()
+		var herr error
+		tID, err := task.GetID()
 		if err != nil {
-			herr := &apierrors.HostConfigError{Msg: err.Error()}
-			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
+			herr = &apierrors.HostConfigError{Msg: err.Error()}
+		} else {
+			err = engine.execCmdMgr.InitializeContainer(tID, container, hostConfig)
+			if err != nil {
+				herr = &apierrors.HostConfigError{Msg: err.Error()}
+			}
 		}
-		err = engine.execCmdMgr.InitializeContainer(tId, container, hostConfig)
-		if err != nil {
-			herr := &apierrors.HostConfigError{Msg: err.Error()}
-			return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(herr)}
+		if herr != nil {
+			seelog.Warnf("Exec Agent initialization: %v . Continuing to start container without enabling exec feature.", herr)
+			container.UpdateManagedAgentByName(execcmd.ExecuteCommandAgentName, apicontainer.ManagedAgentState{
```

Comment:
can you add a TODO to emit managed agent state change event. this is dependent on https://github.com/aws/amazon-ecs-agent/pull/2758